src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -3797,10 +3796,24 @@ bool AdaptDispatch::DoXtermJsAction(const std::wstring_view string)
         // VsCode is using cursorIndex and replacementIndex, but we aren't currently.
         if (succeeded)
         {
-            _api.InvokeMenu(parts.size() < 5 ? L"" : til::at(parts, 4),
-                            replacementLength);
+            // Get the combined lengths of parts 0-3, plus the semicolons. We
+            // need this so that we can just pass the remainder of the string.
+            const auto prefixLength = til::at(parts, 0).size() + 1 +
```

Comment:
I'm mostly horrified that SplitString doesn't have a count limit (but you don't need to add one)